### ISSUE TITLE
Suggestion: Transmogs for cross-armor type

### ISSUE THREAD

**Issue by Aldori15** [azerothcore/mod-transmog#142]
Would it be possible to implement the ability to learn transmogs for different armor types? For example, playing on a Plate character can learn transmogs for Cloth or Leather when they are looted.

Instead of only Plate-->Plate, Mail-->Mail, Leather-->Leather, Cloth-->Cloth.

Maybe this could be done via a setting in the config for people to turn on/off if they didn't want it?

**Comment by Helias** [comment]
there is already a conf about this

`Transmogrification.AllowMixedArmorTypes = 0`

does make what you are looking for?

About sharing the appearance between characters of the same account I think it's another feature request that deserves a different issue

**Comment by Aldori15** [comment]
Yes, I have Transmogrification.AllowMixedArmorTypes = 1, but it's a bit different than what I'm looking for.  That config option allows the ability to transmog different armor types after they are already learned.

But what i'm looking for is the ability to learn appearances no matter what armor type.  If i'm playing on a Plate character and a piece of Cloth armor drops, I can't learn that appearance unless i'm on a Cloth character.

**Comment by Ceikry** [comment]
Possibly resolved by #158

**Comment by Helias** [comment]
@Aldori15 if you could verify that the issue does not persist anymore with the new configuration please let us know and if so, close the issue

**Comment by Aldori15** [comment]
@Helias Will do.  I'll pull down the latest and let you know.

**Comment by Aldori15** [comment]
@Helias The new "Transmogrification.AllowLowerTiers" config option is exactly what we were looking for.  Playing on a Warrior (plate class) and a Cloth piece drops.  Once I equip the Cloth piece, it learns that appearance.  Great, thanks!

- Is it possible to have this functionality also work for higher tiers as well?  For example "Transmogrification.AllowHigherTiers".  Pretty much the exact opposite.  If you're on a Cloth character and a piece of Plate/Mail/Leather drops, would it be possible to learn those appearances as well?

- Is it also possible to do the same thing for the different weapon types as well?  For example Rogues who can't equip staves, but could possibly learn the staff appearance if they loot it?

I'm not sure if these are even possible or not given you currently have to equip the piece of armor in order to learn the appearance.  And since you can't equip higher tier armor pieces, I'm not sure if it's in the realm of possibility, but figured I would ask anyway since you guys know more than I do.

**Comment by Ceikry** [comment]
I can potentially visit the idea of an auto-unlock-on-loot toggle in a future MR, so that you could do it that way. Fair warning though, you would unlock a TON of appearances that way that you probably wouldn't care about.

**Comment by Aldori15** [comment]
Yeah, gm command wouldn't work.  If it's not possible, then I wouldn't sweat it much.  The setting you guys have implemented fixes about half the problem.  If you want to visit the idea of an auto-unlock-on-loot toggle, that would be awesome.  

I'm not too worried about unlocking a bunch of appearances.  The whole idea really is being able to run dungeon/raid content on any class armor type and if loot drops that isn't your armor type, you can still learn those transmog appearances for other characters.  That way you don't have to run the same dungeon on 4 different armor types to get all of the appearances.
